File: Frameworking/ViewController.m
//
//  ViewController.m
//  Frameworking
//
//  Created by Randy McLain on 5/5/16.
//  Copyright © 2016 Randy McLain. All rights reserved.
//

#import "ViewController.h"
#import "Frameworking-Swift.h"
#include "GSPAN.h"


@interface ViewController ()
@property (weak, nonatomic) IBOutlet UISegmentedControl *segmentedControl;
@property (weak, nonatomic) IBOutlet UIButton *button;

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  
  
  
  
  // Do any additional setup after loading the view, typically from a nib.
}
- (IBAction)segmentedSelected:(id)sender {
  [[RestController getInstance] startupServices:@"FitBit"];
}

- (IBAction)buttonPressed:(id)sender {

  [[RestController getInstance] getData:@"FitBit.steps" completion:^(NSArray<NSDictionary<NSString *,id> *> * _Nullable data, NSError * _Nullable error) {
    if (data != nil ) {
      // create an array of like 400 objects to save to core data.  
      NSMutableDictionary *stepsDict = nil;
      
      for (int i =0; i < 366; i ++ ) {
        
        int startDate = 20150101;
        int theDate = (startDate + i);
        int startStep = 0;
        int theStep = (startStep + i);
        
        NSString *dateString = [NSString stringWithFormat:@"%d", theDate];
        NSString *stepString = [NSString stringWithFormat:@"%d", theStep];
        [stepsDict setValue: stepString forKey:dateString];
      }
      
    }
  }];
  
}


- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

@end

File: swift-objc-cpp/OldSkoolNewSkoolDemo/OldSkoolNewSkoolDemo/ViewController.swift
//
//  ViewController.swift
//  OldSkoolNewSkoolDemo
//
//  Created by Randy McLain on 4/2/17.
//  Copyright © 2017 Randy McLain. All rights reserved.
//

import UIKit
import OldSkoolNewSkool

class ViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // I know this isnt doing anything, but the important thing here is that the swift 
        WrapperClass.containsCharacters("this is a string", in: "OMG I cant believe the modules are talking and this is a string")
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

